crates/e2e/tests/e2e/colocation_eth_safe.rs

```diff
@@ -0,0 +1,110 @@
+    let [token] = onchain
+        .deploy_tokens_with_weth_uni_v2_pools(to_wei(1000), to_wei(1000))
+        .await;
+
+    token.mint(trader.address(), to_wei(4)).await;
+    tx!(
+        trader.account(),
+        token.approve(onchain.contracts().allowance, to_wei(4))
+    );
+
+    tracing::info!("Starting services.");
+    let solver_endpoint = colocation::start_solver(onchain.contracts().weth.address()).await;
+    colocation::start_driver(onchain.contracts(), &solver_endpoint, &solver);
+
+    let services = Services::new(onchain.contracts()).await;
+    services.start_autopilot(vec![
+        "--enable-colocation=true".to_string(),
+        "--drivers=test_solver|http://localhost:11088/test_solver".to_string(),
+    ]);
+    services
+        .start_api(vec!["--enable-eth-smart-contract-payments=true".to_string()])
+        .await;
+
+    tracing::info!("Placing order");
+    let balance = onchain
+        .contracts()
+        .weth
+        .balance_of(safe.address())
+        .call()
+        .await
+        .unwrap();
+    assert_eq!(balance, 0.into());
+    let order = OrderCreation {
+        sell_token: token.address(),
+        sell_amount: to_wei(4),
+        buy_token: BUY_ETH_ADDRESS,
+        buy_amount: to_wei(3),
+        valid_to: model::time::now_in_epoch_seconds() + 300,
+        partially_fillable: true,
+        kind: OrderKind::Sell,
+        receiver: Some(safe.address()),
+        ..Default::default()
+    }
+    .sign(
+        EcdsaSigningScheme::Eip712,
+        &onchain.contracts().domain_separator,
+        SecretKeyRef::from(&SecretKey::from_slice(trader.private_key()).unwrap()),
+    );
+    let uid = services.create_order(&order).await.unwrap();
+
+    tracing::info!("Waiting for trade.");
+    let trade_happened = || async {
+        let safe_balance = web3.eth().balance(safe.address(), None).await.unwrap();
+        // the balance is slightly less because of the fee
+        (3_899_000_000_000_000_000_u128..4_000_000_000_000_000_000_u128)
+            .contains(&safe_balance.as_u128())
+    };
+    wait_for_condition(TIMEOUT, trade_happened).await.unwrap();
+
+    onchain.mint_blocks_past_reorg_threshold().await;
+
+    let settlement_event_processed = || async {
+        onchain.mint_block().await;
+        let order = services.get_order(&uid).await.unwrap();
+        if let OrderClass::Limit(LimitOrderClass {
+            executed_surplus_fee,
+        }) = order.metadata.class
+        {
+            executed_surplus_fee > U256::zero()
+        } else {
+            panic!("order is not a limit order");
+        }
+    };
+    wait_for_condition(TIMEOUT, settlement_event_processed)
+        .await
+        .unwrap();
+
+    let tx_hash = services.get_trades(&uid).await.unwrap()[0].tx_hash.unwrap();
+    let competition = services.get_solver_competition(tx_hash).await.unwrap();
+    assert!(!competition.common.solutions.is_empty());
+    assert!(competition.common.auction.orders.contains(&uid));
```

Comment:
Is this required for the test? I think we are testing solver competition endpoints in other test cases and would keep this one to simply asserting that the safe receive some funds in a reasonable time.